planner/core/optimizer.go

```diff
@@ -83,9 +84,25 @@ var optRuleList = []logicalOptRule{
 	&columnPruner{}, // column pruning again at last, note it will mess up the results of buildKeySolver
 }
 
+type logicalOptimizeOp struct {
+	enableTrace bool
+}
+
+type logicalOptimizeOption func(*logicalOptimizeOp)
+
+func withEnableOptimizeTrace() logicalOptimizeOption {
+	return func(op *logicalOptimizeOp) { op.enableTrace = true }
+}
+
```

Comment:
Why do we need this function, it does not seem to be necessary at the moment. What will be added here later?

Reply:
`withEnableOptimizeTrace ` is used in `rule.optimize`. This lets each rule knows whether it needs to collect trace information.

Reply:
It seems that there is not much difference between using `logicalOptimizeOp`. why do we use function parameters?

Reply:
We are using [functional options](https://www.sohamkamani.com/golang/options-pattern/) here